pandas/core/apply.py

```diff
@@ -403,11 +403,18 @@ def agg_dict_like(self) -> DataFrame | Series:
             and selected_obj.columns.nunique() < len(selected_obj.columns)
         )
 
+        # Numba Groupby engine/engine-kwargs passthrough
+        kwargs = {}
+        if is_groupby:
+            engine = self.kwargs.get("engine", None)
+            engine_kwargs = self.kwargs.get("engine_kwargs", None)
+            kwargs = {"engine": engine, "engine_kwargs": engine_kwargs}
```

Comment:
Can we update the method signatures with by adding `engine` and `engine_kwargs` params instead of mangling kwargs?

Reply:
``obj`` here can be a Series/DF/Resampler/ too, since the code is shared with there, so it wouldn't be possible to add ``engine``/``engine_kwargs`` there (unless we supported numba there too), since ``agg`` there is part of the public API.

Reply:
``obj`` here can be a Series/DF/Resampler/ too, since the code is shared with there, so it wouldn't be possible to add ``engine``/``engine_kwargs`` there (unless we supported numba there too), since ``agg`` there is part of the public API.

Reply:
The more we evolve groupby, the more I realize trying to share this code between Series/DataFrame and groupby was a mistake of mine. I've been thinking we should refactor and separate off groupby entirely (but it can still live in `core.apply`). I'm okay with having a bit of a kludge here and will separate/cleanup after.